deps/blake2/CMakeLists.txt

```diff
@@ -1,18 +1,13 @@
 cmake_minimum_required(VERSION 3.22...3.25)
 
-add_library(blake2 STATIC EXCLUDE_FROM_ALL )
+add_library(blake2 OBJECT)
 add_library(OBS::blake2 ALIAS blake2)
 
-target_sources(blake2 PRIVATE src/blake2b-ref.c src/blake2.h src/blake2-impl.h)
+target_sources(
+  blake2
+  PRIVATE src/blake2-impl.h src/blake2b-ref.c
+  PUBLIC src/blake2.h)
 
 target_include_directories(blake2 PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/src")
-set_source_files_properties(
-  src/blake2b-ref.c PROPERTIES COMPILE_OPTIONS
-                               "$<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-error=implicit-int-conversion>")
-
-if(OS_WINDOWS)
-  target_compile_options(blake2 PRIVATE $<IF:$<CONFIG:Debug>,/MTd,/MT> /Zl)
-  target_compile_definitions(blake2 PRIVATE inline=_inline restrict=__restrict)
```

Comment:
I've removed them optimistically and have met no compile or link errors when building the project.

From the little information I was able to dig up, this might have been necessary for quite ancient MSVC versions.

Reply:
I took a quick look and that seems to be the case. It seemed to define `inline` as `_inline` and `restrict` as `__restrict`? MSVC should support `inline` just fine these days, and blake2-impl.h does its own detection for `inline` support, which shows it setting `BLAKE2_INLINE` to `inline` in VS2022 17.7.

I don't see any uses of `restrict` in the BLAKE2 files.

@notr1ch ?

https://github.com/obsproject/obs-studio/commit/23a115b0b3f8215cc63799ab6256c4f28872230c#diff-809b55eb4b1630c733648be54d6d7c9b934eb0671e97ede6b82506ca0481242fR18-R20

Reply:
This is probably fine.